pkg/cache/cache.go

```diff
@@ -117,38 +118,46 @@ func (c *Cache) WaitForPodsReady(ctx context.Context) {
 		return
 	}
 
-	c.Lock()
-	defer c.Unlock()
-
 	log := ctrl.LoggerFrom(ctx)
+
+	podsReadyCond := func(ch chan struct{}) {
+		c.Lock()
+		c.podsReadyCond.Wait()
+		c.Unlock()
+		close(ch)
```

Comment:
```suggestion
		c.Lock()
		defer c.Unlock()
		defer close(ch)		
		c.podsReadyCond.Wait()
```

@stuton Should we use `defer` to catch panic?

Reply:
Yeah, you right